src/wstore/asset_manager/views.py

```diff
@@ -38,29 +40,36 @@
 
 class AssetCollection(Resource):
 
-    @authentication_required
     def read(self, request):
         """
         Retrives the existing digital assets associated with a given seller
         :param request:
         :return: JSON List containing the existing assets
         """
 
+        user = request.GET.get('user', None)
+
         pagination = {
-            'start': request.GET.get('start', None),
-            'limit': request.GET.get('limit', None)
+            'offset': request.GET.get('offset', None),
+            'size': request.GET.get('size', None)
         }
-        if pagination['start'] is None or pagination['limit'] is None:
+        if pagination['offset'] is None or pagination['size'] is None:
             pagination = None
 
-        profile = request.user.userprofile
-
-        if 'provider' not in profile.get_current_roles():
-            return build_response(request, 403, 'You are not authorized to retrieve digital asset information')
+        if user is None:
+            if request.user.is_anonymous():
```

Comment:
This is what the decorator authentication_required does. It doesn't make sense that you remove the decorator and then include the validation in the code.

For consistency with the single asset API, I would allow to make requests to this API without authentication since this is not private information